### ISSUE TITLE
[Container Request]: tb-profiler

### ISSUE THREAD

**Issue by erinyoung** [StaPH-B/docker-builds#416]
### What tool are you looking for and why?

This would be useful to put in a container : https://github.com/jodyphelan/TBProfiler

It doesn't look like the authors have created a container. It is currently available via biocontainers on quay.

**Comment by jwarnn** [comment]
I looked into this tool a while back in May, and I tried to build a dockerfile that manually download all the software dependancies. There is alot, ```trimmomatic (>=v0.38), bwa (>=v0.7.17), minimap2 (>=v2.16), bowtie2 (>=v2.3.5), samtools (>=v1.10), bcftools (>=v1.10), GATK (>=v4.1.4.0), freebayes (>=v1.3.2), tqdm (>=v4.32.2) parallel (>=v20190522) samclip (>=v0.4.0) and snpEff (>=v5.0.0)``` and I got no where fast.  Anyway they have a conda package and a pip3 package that seem to work well. I will append little working versions of each dockerfile and will preform a PR with which ever one is thought to be best. I could also try to install everything manually but that might be a large task. 

```
FROM --platform=linux/x86_64 mambaorg/micromamba:0.24.0 as app

USER root
WORKDIR /

RUN apt-get update && apt-get install -y --no-install-recommends \
 wget \
 ca-certificates \
 procps && \
 apt-get autoclean && rm -rf /var/lib/apt/lists/*

 RUN micromamba install --yes --name base --channel conda-forge --channel bioconda  \
     tb-profiler && \
     micromamba clean --all --yes


 ARG MAMBA_DOCKERFILE_ACTIVATE=1
 WORKDIR /data

FROM app as test

RUN wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR166/009/ERR1664619/ERR1664619_1.fastq.gz && \
    wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR166/009/ERR1664619/ERR1664619_2.fastq.gz && \
    tb-profiler profile -1 ERR1664619_1.fastq.gz -2 ERR1664619_2.fastq.gz -t 4 -p ERR1664619 --txt

```

```
FROM --platform=linux/x86_64 ubuntu:bionic as app

ENV DEBIAN_FRONTEND noninteractive

RUN apt-get update -y && apt-get install -y \
    build-essential \
    zlib1g-dev \
    libncurses5-dev \
    libgdbm-dev \
    libnss3-dev \
    libssl-dev \
    python3-pip \
    libreadline-dev \
    libffi-dev \
    libsqlite3-dev \
    libbz2-dev \
    liblzma-dev \
    git \
    wget && \
    rm -rf /var/lib/apt/lists/* && apt-get autoclean

RUN pip3 install git+https://github.com/jodyphelan/TBProfiler.git && \
    pip3 install git+https://github.com/jodyphelan/pathogen-profiler.git && \
    tb-profiler update_tbdb

WORKDIR /data

FROM app as test

RUN wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR166/009/ERR1664619/ERR1664619_1.fastq.gz && \
    wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR166/009/ERR1664619/ERR1664619_2.fastq.gz && \
    tb-profiler profile -1 ERR1664619_1.fastq.gz -2 ERR1664619_2.fastq.gz -t 4 -p ERR1664619 --txt

```

**Comment by kapsakcj** [comment]
It is a beast of a tool to install, I'd probably prefer the `micromamba` base image route, and let the wonderful bioconda folks handle package management & functionality.

Alternatively we can install the tools manually, as we have dockerfiles for most if not all of the dependencies, but yes it will take some work to put together. Just a lot of copy/paste and tweaking to get correct.

CC @frankambrosio3

**Comment by jodyphelan** [comment]
Just followed a few github issues and found my way here, hope you don't mind!
Just thought I'd mention that not all the dependencies are actually used at run time. For example the user can currently choose from freebayes, gatk, bcftools, pilon and lofreq for variant calling. But I guess if it is packaged into a workflow, only one of them will ever be used and so we can get rid of some deps.

**Comment by jwarnn** [comment]
We don't mind at all. Staph-B is a group of Public Health Bioinformaticians; there has been some interest at Public Health Labs to start surveillance of TB. TBProfile could help get those programs off the ground. So thank you for being responsive to your Git Repository, helping us out, and making a great tool.

**Comment by kapsakcj** [comment]
Hi @jodyphelan ! We appreciate you chiming in and of course for developing & maintaining tb-profiler!

Our goal for our containers is to allow for full functionality of tools if possible, even if it means a bit of extra work to get all tools installed. Thankfully the bioconda package does most if not all of the legwork for us.

I saw your comment about the versioning with the git commit: https://github.com/jodyphelan/tbdb/issues/44#issuecomment-1220489529

Generally we just would like some way to be able to reproducibly install/set up the compute environment that is static/unchanging which includes the database. A GitHub release is ideal since it often includes some version number v.X.X.X (and usually release notes/a changelog), but a git commit works just fine too 👍

**Comment by kapsakcj** [comment]
Thanks for the PR @jwarnn ! Docker image is now available on dockerhub and quay
https://hub.docker.com/r/staphb/tbprofiler/tags
https://quay.io/repository/staphb/tbprofiler?tab=tags

closed via PR #434
